src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -272,10 +277,10 @@ string EncodeUrl (Uri url)
 				if(Logger.LogNet)
 					Logger.Log (LogLevel.Info, LOG_APP, " cancelled");
 
-				httpConnection.Disconnect ();
+				await DisconnectAsync (httpConnection);
```

Comment:
If the code *after* this line is supposed to run on the UI thread then this is correct. If the code is *not* required to run on the UI thread then it should use a `.ConfigureAwait (false)`